pkg/core/blockchain.go

```diff
@@ -25,6 +25,8 @@ var (
 	genAmount         = []int{8, 7, 6, 5, 4, 3, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1}
 	decrementInterval = 2000000
 	persistInterval   = 1 * time.Second
+
+	BlockchainDefault *Blockchain
```

Comment:
it can't be exported

Reply:
What do you mean it can't be exported? I Do need to register a blockchain as a default one in order to use it in the [`References` ](https://github.com/CityOfZion/neo-go/pull/135/commits/c28089009c7419d988cc81693be45f7679806372#diff-3e5ea9b860f32f0d4d6b23ea25baf600R14) method in the fee.go file.
```go

func References(t *transaction.Transaction) map[util.Uint256]*transaction.Output {
	references := make(map[util.Uint256]*transaction.Output)

 	for prevHash, inputs := range t.GroupInputsByPrevHash() {
		if BlockchainDefault == nil {
			panic("no default blockchain available! please register one.")
		} else if tx, _, err := BlockchainDefault.GetTransaction(prevHash); err != nil {
			tx = nil
		} else if tx != nil {
			for _, in := range inputs {
				references[in.PrevHash] = tx.Outputs[in.PrevIndex]
			}
		} else {
			references = nil
		}
	}
	return references
}
```

At least within the `core` package, I do not have any problem to use it `BlockchainDefault`.

Reply:
data race in runtime...

Reply:
it's a bad practice, to set globalVariable..

Reply:
https://dave.cheney.net/2017/06/11/go-without-package-scoped-variables

Reply:
thanks for the link!